### DISCUSSION TITLE
[QUESTION] Slow CTB files for Saturn 2

### DISCUSSION THREAD

**Discussion by Ged91** [sn4k3/UVtools#588] [Q&A]
Hi. I recently switched to an Elegoo Saturn 2 and I noticed that .ctb files are incredibly slow to process in UV Tools. I use it to fix islands etc and I went from 10 minutes to over an hour of processing time. I know that a faster laptop would improve it, but I was wondering if there's anything I can do to improve this. CTB files are a lot heavier than what I was used to with the Photon Mono (like, 10x).

Thanks

**Comment by sn4k3** [comment]
```
Saturn 2   : 7680 x 4320 [33177600 pixels]
Photon Mono: 1620 x 2560 [4147200 pixels]
(7680*4320) / (1620*2560) = Saturn 2 have **8** times more pixels than Photon Mono
```

By math it will be 8x slower and complexer to process Saturn 2 files over Photon Mono, but time does't escalate linear by the pixels and then it's normal that you have a bit more than 8x, in your case you spot about 10x and it's normally fine.

There are nothing you can do to speed up things, the larger the images the slower it will be. Today printers are getting larger and larger resolutions, so you are stuck with what you have or you need to buy a new machine to process this stuff. And if you want to go serious you need a Desktop, no one in sane mind use laptops for heavy workloads and computations.
UVtools is no different than a video editing software, if you want rendering speed you need the best hardware you can afford!

The only thing you can try is to opt for LZ4 layer compression method on Settings, it will gain some time but won't do magic.
Also make sure you are in the latest version

**Reply by Ged91** [reply]
Thank you for the thorough explanation. I wasn't sure it was just a matter of pure size but I guess new hardware it is! (or more patience, much cheaper)

I'll give LZ4 a try. I guess there's no downside to it?

**Reply by sn4k3** [reply]
> I'll give LZ4 a try. I guess there's no downside to it?

It can have the downside when loading ZIP files without printer resolution UVtools is unable to sanitize file resolution. But it only affect ZIP/CWS with that information missing, other than that i recommend it as default: https://github.com/sn4k3/UVtools/discussions/542
